### PR TITLE
Fix list releases panic

### PR DESCRIPTION
**Pull request by marians**
Ensure that https://github.com/giantswarm/gsctl/issues/181 is fixed

This adds a test that shows the panic is not occuring.

In case the API endpoint is not responding, a user now sees

```
$ gsctl list releases
The API didn't send a response.
Please check your connection using 'gsctl ping'. If your connection is fine,
please try again in a few moments.
```

### CODE REVIEW

File: commands/list_releases.go [hunk 1]
```diff
@@ -64,32 +64,35 @@ func init() {
...
-		fmt.Println(color.RedString(err.Error()))
-		os.Exit(1)
+	if err == nil {
+		return
 	}
+
+	handleCommonErrors(err)
+
+	fmt.Println(color.RedString(err.Error()))
+	os.Exit(1)
```
Comment (marians) [inline commands/list_releases.go]: Line 73 to 80 are unrelated to the cause. It's alignment with our command blueprint.

File: commands/list_releases.go [hunk 2]
```diff
@@ -31,8 +31,8 @@ var (
 		Long: `Prints detail on all available releases.
 
 A release is a software bundle that constitutes a cluster. It is identified by its semantic version number.`,
-		PreRun: listReleasesValidationOutput,
-		Run:    listReleasesOutput,
+		PreRun: listReleasesPreRunOutput,
+		Run:    listReleasesRunOutput,
```
Comment (marians) [inline commands/list_releases.go]: Function renaming is boyscouting, unrelated to the PR goals

**Comment by marians** [comment]
Re-pinging reviewers 💐

File: commands/list_releases_test.go [hunk 3]
```diff
@@ -43,6 +43,31 @@ func Test_ListReleases_Empty(t *testing.T) {
...
+	dir, err := tempConfig("")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.RemoveAll(dir)
+
+	// needed to prevent search for the default cluster
+	args := listReleasesArguments{}
+	args.apiEndpoint = "http://localhost:45454"
+	args.token = "my-token"
```
Comment (TheoBrigitte) [inline commands/list_releases_test.go]: Why not ?
```golang
args := listReleasesArguments{
  apiEndpoint: "http://localhost:45454",
  token: "my-token",
}
```

Comment (marians) [reply commands/list_releases_test.go]: Good point, since I just changed this in another PR

**Comment by marians** [comment]
@TheoBrigitte If this looks good to you, can you please give an approval? Thanks

**Pull request merged**